### ISSUE TITLE
Can we add custom models in Tabular Prediction

### ISSUE THREAD

**Issue by EternalConfession** [awslabs/autogluon#1214]
It seems that we can only use "TabularNeuralNetModel" in Tabular Prediction.

Could we use custom deep learning model in this task?

**Comment by Innixma** [comment]
Yes, AutoGluon is fully modular and a custom model can be added as an additional model to train.

While the instructions to do so are not fully documented, you can find a basic tutorial showcasing adding NaiveBayes here: https://github.com/awslabs/autogluon/blob/master/examples/tabular/example_custom_model_tabular.py

To get a better idea of how current models are implemented, you can refer to the model code: https://github.com/awslabs/autogluon/tree/master/tabular/src/autogluon/tabular/models

In future, we will plan to have more comprehensive documentation on this.

**Comment by EternalConfession** [comment]
Thanks for your information

Best Regards,
Jianhui

**Comment by EternalConfession** [comment]
Just another question upon custom models.

Can I specify some parameters sush as learning rate for a single model, while training custom model alongside other models using TabularPredictor.

The senario is that I have a well tuned model, and I want to ensemble it with the autostack models.

Best, 
Jianhui

**Comment by EternalConfession** [comment]
Just some other questions:

###  1. How can we specify a custom evaluation metric?

After reading the example for NaiveBayes, it seems that the evaluation metric is independent of loss in the fit method.

So I can actually specify a bunch of evaluation metrics, and use one as the default metric for validation.

### 2. Can we transfer e.g. PyTorch or TF models to gluon models

### 3. Is there any WebUI for monitor the training process?

**Comment by EternalConfession** [comment]
How can I see the exact hyperparameter space.

For example, for GBM. 
```
{
...
    'GBM': [ {'extra_trees': True, 'ag_args': {'name_suffix': 'XT'}}],
...
}
```

**Comment by EternalConfession** [comment]
{
'GBM': [
    {'extra_trees': True, 'ag_args': {'name_suffix': 'XT'}},
}

What is the effect for the key 'ag_args' and 'extra_trees' ?

As I can find the default hyperparameter searchspace in the implementation of models.

**Comment by EternalConfession** [comment]
After reading the model source code, I have understood most of the problems.

Thanks

**Comment by EternalConfession** [comment]
How can I set the trial numbers for hyperparameter search for each model.

**Comment by EternalConfession** [comment]
Where can I see the default scheduler_params.

**Comment by Innixma** [comment]
Hi @EternalConfession,

Regarding the answer to some of your questions:

In 0.3 release (coming very soon) we will have tutorials available covering:

1. [Adding custom models to AutoGluon](https://auto.gluon.ai/dev/tutorials/tabular_prediction/tabular-custom-model.html)
2. [Adding custom evaluation metrics to AutoGluon](https://auto.gluon.ai/dev/tutorials/tabular_prediction/tabular-custom-metric.html)

Additionally:

1. `ag_args` is for parameters such as model names internally in AutoGluon, generally you shouldn't need to worry about them.
2. `extra_trees` is a hyperparameter of LightGBM.
3. "Can we transfer e.g. PyTorch or TF models to gluon models" -> Any model can be added to AutoGluon via the custom model tutorial. AutoGluon is framework agnostic, so it works with PyTorch and TF models so long as their dependencies are installed.
4. `scheduler_params` can be found in [scheduler_factory](https://github.com/awslabs/autogluon/blob/master/core/src/autogluon/core/scheduler/scheduler_factory.py), but this is not user-facing logic.
